src/Language/PureScript/Interactive/Types.hs

```diff
@@ -54,17 +64,47 @@ allImportsOf m PSCiState{psciImportedModules = is} =
 
 -- * State helpers
 
+updateImportExports :: PSCiState -> PSCiState
```

Comment:
Could you add a comment here saying something like

> This function updates the Imports and Exports values in the PSCiState, which are used for handling completions. This function must be called whenever the PSCiState is modified to ensure that completions remain accurate.

Reply:
Alternatively, is it worth using an explicit export list and hiding the `PSCiState` constructor so that we can easily verify that the `PSCiState` can't get out of sync?

Reply:
Yeah, I think both are good ideas.